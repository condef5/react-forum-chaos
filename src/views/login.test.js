import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Login from "./login";

test("Login component", () => {
  const createUser = jest.fn();

  const { asFragment, getByLabelText } = render(
    <Login createUser={createUser} />
  );

  expect(asFragment()).toMatchSnapshot();

  const inputUsername = getByLabelText("Enter your username");

  fireEvent.change(inputUsername, { target: { value: "test-user" } });

  const inputEmail = getByLabelText("Enter your email");
  fireEvent.change(inputEmail, {
    target: { value: "test-user@mail.com" }
  });

  fireEvent.submit(inputEmail);
  fireEvent.submit(inputUsername);

  expect(createUser).toHaveBeenCalledWith({
    username: "test-user",
    email: "test-user@mail.com"
  });
});
