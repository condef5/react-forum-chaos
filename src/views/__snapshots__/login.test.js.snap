// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component 1`] = `
<DocumentFragment>
  .emotion-6 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  background-color: white;
  border-radius: .5rem;
  padding: 2rem 1rem;
  width: 50%;
  max-width: 360px;
  margin: auto;
}

.emotion-0 {
  margin: 0;
  font-weight: 500;
  text-align: center;
}

.emotion-2 {
  margin: 2.5rem 1rem;
}

.emotion-1 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  width: 100%;
}

.emotion-1:focus {
  outline: none;
  border-color: #444444;
}

.emotion-5 {
  background-color: black;
  border: 1px solid black;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-5:hover {
  background-color: white;
  color: black;
}

<div
    class="emotion-6"
  >
    <form>
      <h1
        class="emotion-0"
      >
        Welcome to Chaos
      </h1>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Enter your username"
          class="emotion-1"
          id="username"
          name="username"
          placeholder="Enter your username"
          type="text"
          value=""
        />
      </div>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Enter your email"
          class="emotion-1"
          id="email"
          name="email"
          placeholder="Enter your email"
          type="email"
          value=""
        />
      </div>
      <button
        class="emotion-5"
        type="submit"
      >
        Enter
      </button>
    </form>
  </div>
</DocumentFragment>
`;
